src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSWebWorker.cs

```diff
@@ -49,20 +53,18 @@ internal sealed class JSWebWorkerInstance<T> : IDisposable
             private Thread _thread;
             private CancellationToken _cancellationToken;
             private CancellationTokenRegistration? _cancellationRegistration;
-            private Func<Task<T>>? _bodyRes;
-            private Func<Task>? _bodyVoid;
-            private Task? _resultTask;
+            private Func<Task<T>> _body;
+            private Task<T>? _resultTask;
             private bool _isDisposed;
 
-            public JSWebWorkerInstance(Func<Task<T>>? bodyRes, Func<Task>? bodyVoid, CancellationToken cancellationToken)
+            public JSWebWorkerInstance(Func<Task<T>> bodyRes, CancellationToken cancellationToken)
```

Comment:
Just curious: couldn't the unification to use always task with result have some negative impact? Like for example `Task.CompletedTask` is cached/singleton. Something like that?

Reply:
I think that part of it is ok.